### PR TITLE
fix: Decouple samtranslator.models and *.intrinsics and add import tests

### PR DESCRIPTION
**Pull request by aahung**
### Description of changes

#2975

`samtranslator.models.__init__` should not import `samtranslator.intrinsics.*` which in turn import `samtranslator.models.__init__`

### Description of how you validated changes

Added a new import test, without the fix, it will fail:

image

### Checklist

- [ ] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [ ] Using correct values
    - [ ] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/schema/schema.json [hunk 1]
```diff
@@ -193550,10 +193550,10 @@
         "InvokeRole": {
           "anyOf": [
             {
-              "type": "string"
+              "type": "object"
```
Comment (acristin) [inline samtranslator/schema/schema.json:193553]: I don't understand why the schema is changing.

Comment (GavinZZ) [reply samtranslator/schema/schema.json:193553]: curious too

Comment (aahung) [reply samtranslator/schema/schema.json:193553]: no idea. in gh action it decided to generate with different order: https://github.com/aws/serverless-application-model/actions/runs/4296629078/jobs/7488579979

Comment (aahung) [reply samtranslator/schema/schema.json:193553]: possibly pydantic orders the keys according to the hash of our module `samtranslator`

**Review by GavinZZ — APPROVED** [review]

**Review by acristin — APPROVED** [review]

**Review by hoffa — APPROVED** [review]

**Pull request merged**